### ISSUE TITLE
Card does not clip image

### ISSUE THREAD

**Issue by lukepighetti** [flutter/flutter#28263]
```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, v1.2.1, on Mac OS X 10.14.3 18D109, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
[✓] Android Studio (version 3.1)
[✓] VS Code (version 1.31.1)
[✓] Connected device (2 available)

• No issues found!
```

```dart

import 'package:flutter/material.dart';

main() {
  runApp(_MyApp());
}

class _MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: Text("Card doesn't clip images")),
        body: Center(
          child: _Card(
            image: NetworkImage(
                "https://dtpmhvbsmffsz.cloudfront.net/posts/2015/05/12/55529027d14d7b1e2f002521/m_55529027d14d7b1e2f002522.jpg"),
            title: "Does not clip upper corners",
          ),
        ),
      ),
    );
  }
}

class _Card extends StatelessWidget {
  final ImageProvider image;
  final String title;

  _Card({@required this.image, @required this.title});

  @override
  Widget build(BuildContext context) {
    return Card(
      clipBehavior: Clip.hardEdge,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: <Widget>[
          Expanded(
            child: Image(
              image: image,
              fit: BoxFit.cover,
            ),
          ),
          Padding(
            padding: EdgeInsets.all(12),
            child: Text(title),
          )
        ],
      ),
    );
  }
}
```

![img_d71308674795-1](https://user-images.githubusercontent.com/666539/53182422-789f9880-35c7-11e9-9fc9-287522619dd9.jpeg)

**Comment by zoechi** [comment]
Seems to work for me 
![image](https://user-images.githubusercontent.com/405837/53184630-ac95b080-35fd-11e9-8481-e0ba52c924fc.png)

```
[✓] Flutter (Channel master, v1.2.2-pre.21, on Mac OS X 10.14.3 18D42, locale en-AT)
```

**Comment by lukepighetti** [comment]
Resolved in 1.2.2?

**Comment by zoechi** [comment]
I don't know the latest version of `dev` channel, but I'm sure it will be included in the next `dev` channel release when it happens.
